src/aead/less_safe_key.rs

```diff
@@ -44,6 +44,26 @@ impl LessSafeKey {
         })
     }
 
+    /// Like [open_in_place](Self::open_in_place), except it accepts an arbitrary authentication tag.
+    #[inline]
+    pub fn open_in_place_seperate_tag<'in_out, A>(
+        &self,
+        nonce: Nonce,
+        aad: Aad<A>,
+        tag: Tag,
+        in_out: &'in_out mut [u8],
+    ) -> Result<&'in_out mut [u8], error::Unspecified>
+    where
+        A: AsRef<[u8]>,
+    {
+        let received_tag = tag;
+        let aad = Aad::from(aad.as_ref());
+
+        let ciphertext_len = in_out.len();
```

Comment:
I think you don't need to pass `ciphertext_len` as a parameter.

Reply:
I had originally passed it to avoid doing more work inside of `open_within_` to calculate something we already had, but I imagine that the optimizer will have an easy enough time eliding things that it shouldn't be an issue.